omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ConditionController2_2Test.java

```diff
@@ -161,6 +163,80 @@ public void shouldCreateACodedCondition() throws Exception {
 		Assert.assertNotNull(newConditionSource.get("onsetDate"));
 	}
 	
+	@Test
+	public void shouldUpdateANonCodedCondition() throws Exception {
+		Condition condition = conditionService.getConditionByUuid(RestTestConstants2_2.NON_CODED_CONDITION_UUID);
+		CodedOrFreeText oldCondition = condition.getCondition();
+		
+		Assert.assertNotNull(condition);
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
+		codedOrFreeText.add("nonCoded", "Updated non coded condition");
+		
+		SimpleObject attributes = new SimpleObject();
+		attributes.add("condition", codedOrFreeText);
+		attributes.add("clinicalStatus", ConditionClinicalStatus.ACTIVE);
+		attributes.add("verificationStatus", ConditionVerificationStatus.CONFIRMED);
+		attributes.add("onsetDate", "2018-02-23 00:00:00");
+		attributes.add("voided", false);
+		
+		String json = new ObjectMapper().writeValueAsString(attributes);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI() + "/"
+		        + RestTestConstants2_2.NON_CODED_CONDITION_UUID);
+		req.setContent(json.getBytes());
+		handle(req);
+		
+		condition = conditionService.getConditionByUuid(RestTestConstants2_2.NON_CODED_CONDITION_UUID);
+		
+		LinkedHashMap cond = attributes.get("condition");
+		
+		Assert.assertNotNull(condition);
+		Assert.assertNotEquals(oldCondition.getNonCoded(), condition.getCondition().getNonCoded());
+		Assert.assertEquals(cond.get("nonCoded"), condition.getCondition().getNonCoded());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, condition.getClinicalStatus());
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, condition.getVerificationStatus());
+		Assert.assertNotNull(condition.getOnsetDate());
+	}
+	
+	@Test
+	public void shouldUpdateACodedCondition() throws Exception {
+		Condition condition = conditionService.getConditionByUuid(RestTestConstants2_2.CODED_CONDITION_UUID);
+		CodedOrFreeText oldCondition = condition.getCondition();
+		
+		Assert.assertNotNull(condition);
+		
+		SimpleObject codedOrFreeText = new SimpleObject();
```

Comment:
Here, you need to add checks for the Update() method. Then why are you creating a new object instead of editing the condition for the existing retrieved object?

Reply:
I hadn't thought of it that way. Nice catch.
Let me change my approach.